tests/ui/rfcs/rfc-0000-never_patterns/bindings.rs

```diff
@@ -0,0 +1,25 @@
+#![feature(never_patterns)]
+#![allow(incomplete_features)]
+
+enum Void {}
+
+fn main() {
+    let x: Result<bool, &(u32, u32, Void)> = Ok(false);
+
+    match x {
+        Ok(_x) | Err(&!) => {}
+    }
+    let (Ok(_x) | Err(&!)) = x;
```

Comment:
how does this handle weird things like:

* `a | !`
* `a | (a, _ | !)`

idk, maybe you could think of weirder patterns